allensdk/api/cloud_cache/cloud_cache.py

```diff
@@ -0,0 +1,305 @@
+import os
+import copy
+import io
+import pathlib
+import pandas as pd
+import boto3
+from botocore import UNSIGNED
+from botocore.client import Config
+from allensdk.internal.core.lims_utilities import safe_system_path
+from allensdk.api.cloud_cache.manifest import Manifest
+from allensdk.api.cloud_cache.file_attributes import CacheFileAttributes  # noqa: E501
+from allensdk.api.cloud_cache.utils import file_hash_from_path  # noqa: E501
+from allensdk.api.cloud_cache.utils import bucket_name_from_uri  # noqa: E501
+from allensdk.api.cloud_cache.utils import relative_path_from_uri  # noqa: E501
+
+
+class CloudCache(object):
+    """
+    A class to handle the downloading and accessing of data served from a cloud
+    storage system
+
+    Parameters
+    ----------
+    cache_dir: str or pathlib.Path
+        Path to the directory where data will be stored on the local system
+    """
+
+    _bucket_name = None
+
+    def __init__(self, cache_dir):
+        self._manifest = Manifest(cache_dir)
+        self._s3_client = None
+        self._manifest_file_names = self._list_all_manifests()
+
+    @property
+    def file_id_column(self) -> str:
+        """
+        The column in the metadata files used to uniquely
+        identify data files
+        """
+        return self._manifest.file_id_column
+
+    @property
+    def version(self) -> str:
+        """
+        The version of the dataset currently loaded
+        """
+        return self._manifest.version
+
+    @property
+    def metadata_flie_names(self) -> list:
```

Comment:
Look like a type? Should be: `def metadata_file_names(self) -> list:`

Reply:
Done